DIPs/DIP8484.md

```diff
@@ -0,0 +1,101 @@
+# Traits omniscience
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 8484                                                            |
+| Author:         | Basile Burg                                                     |
+| Implementation: |                                                                 |
+| Status:         | Draft                                                           |
+
+## Abstract
+
+The results of `__traits()` are limited by the protection attributes.
+
+This DIP proposes to remove the limitation from all the traits, except `allMembers` that will get a new parameter allowing it to be omniscient.
+
+### Links
+
+- https://forum.dlang.org/thread/nq512d$9po$1@digitalmars.com
+- https://issues.dlang.org/show_bug.cgi?id=15335
+- https://issues.dlang.org/show_bug.cgi?id=15371
+
+## Description
+
+### New trait parameter
+
+A second optional trait parameter, a `bool` will be added to the trait `allMembers`.
+
+`__traits(allMember, Parent, false)` will be strictly the same as `__traits(allMember, Parent)`. 
+`__traits(allMember, Parent, true)` will toggle omnisciency on and each member of `Parent`, wether it would be accessible or not in a standard statement, will be in the resulting tuple.
+
+It's also worth mentioning that `getProtection` can be used to stay compliant with the protection attributes, despite of the super visual acuity obtained when the new parameter is set to `true`.
+
+### Changes applied to the existing traits
+
+The new `allMembers` parameter only allows to get the identifiers of the protected members. To make the results usable, the protection checks must be removed from each particular trait, including:
+
+- `getMember`
+- `getOverloads`
+- `derivedMembers`
+- (TODO: complete the list)
+
+It doesn't matter if they see everything because `allMembers` without the new default parameter acts as a filter.
+
```

Comment:
paragraph to specify how this would mix with safety, (see remark about [access to `private @trusted` members](https://forum.dlang.org/post/byqyquckgckmyqwtqwis@forum.dlang.org))